backend/portal/logic.py

```diff
@@ -35,25 +35,32 @@ def get_user_populations(user):
 
     user_info = get_user_info(user)
 
-    year = [
-        TargetPopulation.objects.get(
-            kind=TargetPopulation.KIND_YEAR, population=user_info["student"]["graduation_year"]
-        )
-    ]
+    year = (
+        [
+            TargetPopulation.objects.get(
+                kind=TargetPopulation.KIND_YEAR, population=user_info["student"]["graduation_year"]
+            )
+        ]
+        if user_info["student"]["graduation_year"]
+        else []
+    )
 
     school = [
         TargetPopulation.objects.get(kind=TargetPopulation.KIND_SCHOOL, population=x["name"])
         for x in user_info["student"]["school"]
+        if user_info["student"]["school"]
```

Comment:
this won't really do anything. I think you want something similar to how you handle `year`. Basically, for each `x`, it is going to check if `user_info["student"]["school"]` exists which we know to be `True` if it is already iterating. Similarly, it'll still throw an error if it doesn't exist.